src/wallet/rpcdump.cpp

```diff
@@ -358,8 +355,7 @@ UniValue importprunedfunds(const JSONRPCRequest& request)
         throw JSONRPCError(RPC_INVALID_ADDRESS_OR_KEY, "Something wrong with merkleblock");
     }
 
-    auto locked_chain = pwallet->chain().lock();
-    Optional<int> height = locked_chain->getBlockHeight(merkleBlock.header.GetHash());
+    Optional<int> height = pwallet->chain().getBlockHeight(merkleBlock.header.GetHash());
```

Comment:
I think the `LOCK(pwallet->cs_wallet);` line below should move above this `getBlockHeight()` call. Otherwise there could be a race condition if there is a reorg and the transaction is no longer confirmed by the time AddToWallet is called below, but the wallet considers it confirmed anyway.

I'm not sure if it would be a good idea, but perhaps the `CWallet::chain()` method should be annotated `EXCLUSIVE_LOCKS_REQUIRED(cs_wallet)` to catch other cases like this.

Reply:
Thanks, I've modified it, but didn't add `EXCLUSIVE_LOCKS_REQUIRED` to the `CWallet::chain()` as it would change lock assumptions for RPC calls like `bumpfee` or `fundtransaction`..